testutils/consul.go

```diff
@@ -50,3 +53,95 @@ func NewTestConsulServer(tb testing.TB, config TestConsulServerConfig) *testutil
 
 	return srv
 }
+
+// Bulk add test data for seeding consul
+func AddServices(t testing.TB, srv *testutil.TestServer, svcs []testutil.TestService) {
+	for _, s := range svcs {
+		RegisterConsulService(t, srv, s, testutil.HealthPassing)
+	}
+}
+
+// Generate service TestService entries.
+// Services with different IDs and Names.
+func TestServices(n int) []testutil.TestService {
+	return generateServices(n,
+		func(i int) string {
+			return fmt.Sprintf("svc_name_%d", i)
+		},
+		func(i int) string {
+			return fmt.Sprintf("svc_id_%d", i)
+		})
+}
+
+// Generate service instance TestService entries.
+// Services with different IDs but with the same name.
+func TestInstances(n int) []testutil.TestService {
+	return generateServices(n,
+		func(i int) string {
+			return fmt.Sprintf("svc_name_common")
+		},
+		func(i int) string {
+			return fmt.Sprintf("svc_id_%d", i)
+		})
+}
+
+// shorter name for the formatting functions
+type fmtFunc func(i int) string
+
+// does the actual work of generating the TestService objects
+func generateServices(n int, namefmt, idfmt fmtFunc) []testutil.TestService {
+	baseport := 30000
+	services := make([]testutil.TestService, n)
+	for i := 0; i < n; i++ {
+		services[i] = testutil.TestService{
+			Name:    namefmt(i),
+			ID:      idfmt(i),
+			Address: "127.0.0.2",
+			Port:    int(baseport + i),
+			Tags:    []string{},
+		}
+	}
+	return services
+}
+
+// ---------------------------------------------------------------------------
+// The below functions are not used in tests but are handy to keep around for
+// when you need to check on Consul's data .
+
+// All registered services
+func ShowMeServices(t testing.TB, srv *testutil.TestServer) {
+	u := fmt.Sprintf("http://%s/v1/agent/services", srv.HTTPAddr)
+	resp := RequestHTTP(t, http.MethodGet, u, "")
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(string(b))
+	defer resp.Body.Close()
+}
+
+// Health status for all services
+func ShowMeHealth(t testing.TB, srv *testutil.TestServer) {
+	// get the node
+	u := fmt.Sprintf("http://%s/v1/health/service/%s", srv.HTTPAddr, "svc_0")
```

Comment:
It looks like "svc_0" is a service name that the engineer would manually modify. If I'm understanding this right, I think it could be useful to make this value a variable so that engineers won't need to come into the method to modify it.

Reply:
Thanks for the suggestion. I wasn't sure about other use cases so I just left them as I had them, but I'm happy to make this change.